language-php.md

```diff
@@ -44,6 +44,15 @@ Yes:
 $variableName
 ```
 
+### Function (Only PHP7)
+We declare function parameter type and return type.
```

Comment:
I kinda not suggest writing return type in some conditions:
- In some function if we don't want to return anything.
- There are more than one type might return in the function, even native php function, many might return two types, some might return more than 4 types.
- If return `null`, not type to hit.

Reply:
- http://php.net/manual/en/migration71.new-features.php#migration71.new-features.void-functions
- I think https://stackoverflow.com/a/45262566/6477262 explain clear than me.
- http://php.net/manual/en/migration71.new-features.php#migration71.new-features.nullable-types

I know there is a place where cannot declare return type, it's anonymous functions for `array_map` or something like that.

Reply:
Here's the thing to notice, if we not declare strict_types = 1, type declaration will not throw error and convert to target type.

Reply:
@to2false 
Nice to know. for 1, 3. no problem now, thanks.

For point 2, I don't think return multiple types is a potential design flaw, because we need it sometimes: 
like the `get` function in the laravel `Cache`, it might return `int` or `string`, or `array` ... base on what we put in the cache,  so the cache `put` function, the parameter type might be `int`, `string` ... as well. similar things like `config`, `env` ... functions.

by the way wen can declare return type in anonymous function, even `array_map`:
```php
$arr = [1, 2, 3];
$arr = array_map(function (int $val): int {
    return $val + 1;
}, $arr);
var_dump($arr);
```